Source/Diagnostics/ReducedDiags/FieldProbe.cpp

```diff
@@ -0,0 +1,280 @@
+
+                    amrex::ParticleReal Ex_interp = 0._rt, Ey_interp = 0._rt, Ez_interp = 0._rt;
+                    amrex::ParticleReal Bx_interp = 0._rt, By_interp = 0._rt, Bz_interp = 0._rt;
+
+                    amrex::Array<amrex::Real, 3> v_galilean = {{0}};
+                    const auto
+                        &xyzmin = WarpX::GetInstance().LowerCornerWithGalilean(box, v_galilean,
+                                                                               lev);
+                    const std::array<Real, 3> &dx = WarpX::CellSize(lev);
+                    amrex::GpuArray<amrex::Real, 3> dx_arr = {dx[0], dx[1], dx[2]};
+                    amrex::GpuArray<amrex::Real, 3> xyzmin_arr = {xyzmin[0], xyzmin[1], xyzmin[2]};
+
+                    //Interpolating to the probe point
+                    doGatherShapeN(x_probe, y_probe, z_probe, Ex_interp, Ey_interp, Ez_interp,
+                                   Bx_interp, By_interp, Bz_interp, arrEx, arrEy, arrEz, arrBx,
+                                   arrBy, arrBz, Extype, Eytype, Eztype, Bxtype, Bytype, Bztype,
+                                   dx_arr, xyzmin_arr, amrex::lbound(box),
+                                   WarpX::n_rz_azimuthal_modes, interp_order, false);
+
+                    // Either save the interpolated fields or the raw fields depending on the raw_fields flag
+                    fp_Ex = raw_fields ? arrEx(i_probe, j_probe, k_probe) : Ex_interp;
+                    fp_Ey = raw_fields ? arrEy(i_probe, j_probe, k_probe) : Ey_interp;
+                    fp_Ez = raw_fields ? arrEz(i_probe, j_probe, k_probe) : Ez_interp;
+                    fp_E = Ex_interp * Ex_interp + Ey_interp * Ey_interp + Ez_interp * Ez_interp;
+                    fp_Bx = raw_fields ? arrBx(i_probe, j_probe, k_probe) : Bx_interp;
+                    fp_By = raw_fields ? arrBy(i_probe, j_probe, k_probe) : By_interp;
+                    fp_Bz = raw_fields ? arrBz(i_probe, j_probe, k_probe) : Bz_interp;
+                    fp_B = Bx_interp * Bx_interp + By_interp * By_interp + Bz_interp * Bz_interp;
+
+                    probe_proc = amrex::ParallelDescriptor::MyProc();
+                }
+            }
+
+            //All the processors have probe_proc = -1 except the one that contains the point, which
+            //has probe_proc equal to a number >=0. Therefore, ReduceIntMax communicates to all the
+            //processors the rank of the processor which contains the point
+            amrex::ParallelDescriptor::ReduceIntMax(probe_proc);
+
+            if(probe_proc != amrex::ParallelDescriptor::IOProcessorNumber()) {
+                if (amrex::ParallelDescriptor::MyProc() == probe_proc) {
+                    amrex::ParallelDescriptor::Send(&fp_Ex,
+                                                    1,
+                                                    amrex::ParallelDescriptor::IOProcessorNumber(),
+                                                    0);
```

Comment:
You can potentially use `index_Ex`, etc. as tags in the MPI calls.

Reply:
Good idea!